include/wx/msw/chkconf.h

```diff
@@ -225,6 +225,14 @@
 #   endif /* wxUSE_ACTIVEX */
 #endif /* __WINE__ */
 
+#if wxUSE_GRAPHICS_CONTEXT && defined(__WXUNIVERSAL__)
```

Comment:
I don't really mind merging this to fix the build, but I'd like to mention that there is no reason wxGC shouldn't be available in wxUniv. I.e. the real fix would be fix the linking error instead.

Reply:
I will see if I can find the cause; do you know the name of the likely library names that are either missing or in the wrong place?

Tim S.

Reply:
No, sorry, but the libraries should be easy to determine from the linker error -- what is it exactly?

Reply:
I will have to remove the fix and rebuild; be at least a few hours to rebuild and find the error.
Tim S.

Reply:
Found the file I saved the error it was src/common/graphcmn.cpp:984: undefined reference to `wxGraphicsRenderer::GetDefaultRenderer()'

I will see if I can find the library name from that.

Tim S.

Reply:
Thinking about it; I am afraid the problem is not likely a missing library but missing code.

Reply:
This is not due to a missing library, but seems to be due to `wxUSE_GRAPHICS_GDIPLUS` being 0, so that this function is not getting defined in `src/msw/graphics.cpp` (the only alternative I see is that this file is not being compiled, but it should be).

So I guess we need to define this one, and probably `wxUSE_GRAPHICS_DIRECT2D` too (because why not) in `wx/univ/setup.h`. But actually I have no idea why we don't include all MSW-specific options in it automatically... probably just because nobody could be bothered to do it.

Reply:
I just edited and forced push my last commit.

Tim S.

Reply:
P.S. If you'd like to make this work reliably, you need to modify `build/update-setup-h` to include the contents of `include/wx/msw/setup_inc.h` into `include/wx/univ/setup.h`, like it's already done for several other files. Currently this is not done and instead we have a quite useless `include/wx/univ/setup_inc.h` which could be just removed and folded directly into `include/wx/univ/setup.h` itself because it's not needed everywhere else. Then any new wxMSW-specific options would be automatically added for wxUniv when `build/update-setup-h` is rerun.

Reply:
I will see about that as a possible long term task; no time right now.
Tim S.

Reply:
Please read my updated comment do I need to add to it?
I missed your post about wxUSE_GRAPHICS_GDIPLUS needing to be defined.
So, do I need to update the comment about that?

Tim S.

Reply:
I figure I should rebuild to verify none of the minor changes breaks the windows build.
Tim S.

Reply:
I just realized that I need to try it since, it appears to be a easy fix if just adding wxUSE_GRAPHICS_GDIPLUS fixes the problem.

Reply:
I just did a forced push I hope I am correct that under Windows "wxUSE_GRAPHICS_CONTEXT requires wxUSE_GRAPHICS_GDIPLUS". I added an error message that states that.

Tim S.